library/protocol_common.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef ELLIPTICS_PROTOCOL_COMMON_HPP
+#define ELLIPTICS_PROTOCOL_COMMON_HPP
+
+#include <elliptics/packet.h>
+#include <elliptics/utils.hpp>
+
+namespace ioremap { namespace elliptics {
+
+#define DNET_READ_FLAGS_JSON (1<<0)
+#define DNET_READ_FLAGS_DATA (1<<1)
+
+static inline const char *dnet_dump_read_flags(uint64_t flags)
+{
+	static __thread char buffer[256];
+	static struct flag_info infos[] = {
+		{ DNET_READ_FLAGS_JSON, "json" },
+		{ DNET_READ_FLAGS_DATA, "data" },
+	};
+
+	dnet_flags_dump_raw(buffer, sizeof(buffer), flags, infos, sizeof(infos) / sizeof(infos[0]));
+
+	return buffer;
+};
+
+data_pointer serialize(const dnet_json_header &value);
```

Comment:
Why are these 2 function added?

Reply:
`dnet_json_header` serialization doesn't belong to net protocol, so it must be accessible without including of header `protocol.hpp`

Reply:
So it requires including `protocol_common.hpp`. What is the difference?

Reply:
`protocol.hpp` will be removed, `protocol_common.hpp` not.

If you think it is too preliminary work, I can remove. I done this changes when I tried to write eblob handlers that use new protocol in separate source file. There I included `n2_protocol.hpp` and didn't include `protocol.hpp`. And code didn't build because `dnet_json_header` was required independently on protocol. I decided to extract that from `protocol.hpp`. We should do it in future anyway. `protocol_common.hpp` maybe isn't good name, but separation is good idea I think.

Reply:
Let's formulate and implement this idea separately from this PR.